pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -993,20 +1004,44 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(ctx context.Context, clusterName string
 		return nil, fmt.Errorf("error getting floating ip for port %s: %v", portID, err)
 	}
 	if floatIP == nil && floatingPool != "" && !internalAnnotation {
-		glog.V(4).Infof("Creating floating ip for loadbalancer %s port %s", loadbalancer.ID, portID)
-		floatIPOpts := floatingips.CreateOpts{
-			FloatingNetworkID: floatingPool,
-			PortID:            portID,
-		}
-
 		loadBalancerIP := apiService.Spec.LoadBalancerIP
+		needCreate := true
+		// check first does floatingip exist already in project, otherwise try to create new one
 		if loadBalancerIP != "" {
-			floatIPOpts.FloatingIP = loadBalancerIP
+			floatingip, err := getFloatingIPByFloatingIP(lbaas.network, loadBalancerIP)
+			if err != nil {
+				glog.V(4).Infof("could not find floating ip %s from project: %v", floatingip, err)
+			} else {
+				if len(floatingip.PortID) == 0 {
+					floatUpdateOpts := floatingips.UpdateOpts{
+						PortID: &portID,
+					}
+					floatIP, err = floatingips.Update(lbaas.network, floatingip.ID, floatUpdateOpts).Extract()
+					if err != nil {
+						return nil, fmt.Errorf("error updating LB floatingip %+v: %v", floatUpdateOpts, err)
+					} else {
+						needCreate = false
+					}
+				} else {
+					return nil, fmt.Errorf("floatingip is attached already to another port")
+				}
+			}
 		}
+		if needCreate {
+			glog.V(4).Infof("Creating floating ip for loadbalancer %s port %s", loadbalancer.ID, portID)
+			floatIPOpts := floatingips.CreateOpts{
+				FloatingNetworkID: floatingPool,
+				PortID:            portID,
+			}
 
-		floatIP, err = floatingips.Create(lbaas.network, floatIPOpts).Extract()
-		if err != nil {
-			return nil, fmt.Errorf("error creating LB floatingip %+v: %v", floatIPOpts, err)
+			if loadBalancerIP != "" {
+				floatIPOpts.FloatingIP = loadBalancerIP
```

Comment:
may be we should add the following, so someone reading logs will understand that we are trying to create one with a floating ip specified in the annotation since we also log the "could not find" from line 1013.  

```
glog.V(4).Infof("creating a new floating ip %s", loadBalancerIP)
```

what do you think?